### PR TITLE
Add Travis support

### PR DESCRIPTION
**Pull request by janisozaur**
Please enable Travis-CI: https://travis-ci.org/ this one is free-for-open-source CI service which allows running Linux and macOS jobs.

Currently the code for macOS is broken, so there is no configuration enabled.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - os: linux
+      env: MATRIX_EVAL="CC=clang-5.0 && CXX=clang++-5.0"
+      addons:
+        apt:
+          sources:
+            - llvm-toolchain-trusty-5.0
+            - ubuntu-toolchain-r-test
+          packages:
+            - clang-5.0
+            - libstdc++-5-dev
```
Comment (janisozaur) [inline .travis.yml]: The default workers come with GCC 4.8 installed, it has too old `libstdc++` for some of the features used here.

File: appveyor.yml [hunk 2]
```diff
@@ -1,9 +1,14 @@
-version: 0.5.{build}
-image:
-- Visual Studio 2017
-- Visual Studio 2015
-build_script:
-- cmd: >-
-    cmake .
```
Comment (nfrechette) [inline appveyor.yml]: I imagine the line endings changed here?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# prevent Travis from overwriting our CXX variables
+before_install:
+    - eval "${MATRIX_EVAL}"
+    - echo $CXX
+
+script:
+    - mkdir build
+    - cd build
+    - cmake --config Debug ..
```
Comment (nfrechette) [inline .travis.yml]: appveyor uses release config but travis debug?

File: appveyor.yml (hunk 2 above)
Comment (janisozaur) [reply appveyor.yml]: yes, i didn't notice they were wrong in the first place :(.

You can add [`?w=1`](https://github.com/nfrechette/acl/pull/17/files?w=1) to GitHub's diff page to see past the whitespace.

File: .travis.yml (hunk 3 above)
Comment (janisozaur) [reply .travis.yml]: Yes, in case of errors in tests, I think the Linux build should give you nice stack trace.

Comment (janisozaur) [reply .travis.yml]: But that's something you would want to verify.

Comment (janisozaur) [reply .travis.yml]: https://ci.appveyor.com/project/janisozaur/acl/build/job/0jr9jr70rxa6au4t and https://travis-ci.org/janisozaur/acl/jobs/306959728

**Pull request merged**